client/memconfig.go

```diff
@@ -0,0 +1,153 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const (
+	memConfigEndpoint = "/memstore/config"
+)
+
+// MemConfig ... contains properties that are used to configure the MemStore's behavior.
+// this is copied directly from /store/generated_key/memstore/memconfig. 
+// importing the struct isn't possible since it'd create cyclic dependency loop 
+// with core proxy's go.mod
+type MemConfig struct {
+	MaxBlobSizeBytes uint64
+	BlobExpiration   time.Duration
+	PutLatency time.Duration
+	GetLatency time.Duration
+	PutReturnsFailoverError bool
+}
+
+// MarshalJSON implements custom JSON marshaling for Config.
+// This is needed because time.Duration is serialized to nanoseconds,
+// which is hard to read.
+func (c MemConfig) MarshalJSON() ([]byte, error) {
+	return json.Marshal(intermediaryCfg{
+		MaxBlobSizeBytes:        c.MaxBlobSizeBytes,
+		BlobExpiration:          c.BlobExpiration.String(),
+		PutLatency:              c.PutLatency.String(),
+		GetLatency:              c.GetLatency.String(),
+		PutReturnsFailoverError: c.PutReturnsFailoverError,
+	})
+}
+
+
+// intermediaryCfg ... used for decoding into a less rich type before
+// translating to a structured MemConfig
+type intermediaryCfg struct {
+	MaxBlobSizeBytes        uint64
+	BlobExpiration          string
+	PutLatency              string
+	GetLatency              string
+	PutReturnsFailoverError bool
+}
+
+// IntoMemConfig ... converts an intermediary config into a memconfig 
+// with structured type definitions
+func (cfg *intermediaryCfg) IntoMemConfig() (*MemConfig, error) {
+	getLatency, err := time.ParseDuration(cfg.GetLatency)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse getLatency: %w", err)
+	}
+
+	putLatency, err := time.ParseDuration(cfg.PutLatency)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse putLatency: %w", err)
+	}
+
+	blobExpiration, err := time.ParseDuration(cfg.BlobExpiration)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse blobExpiration: %w", err)
+	}
+
+	return &MemConfig{
+		MaxBlobSizeBytes: cfg.MaxBlobSizeBytes,
+		BlobExpiration: blobExpiration,
+		PutLatency: putLatency,
+		GetLatency: getLatency,
+		PutReturnsFailoverError: cfg.PutReturnsFailoverError,
+	}, nil
+}
+
+
+// MemCfgClient implements a standard client for the eigenda-proxy
+// that can be used for updating a memstore configuration in real-time
+// this is useful for API driven tests in protocol forks that leverage
+// custom integrations with EigenDA
+type MemCfgClient struct {
+	cfg        *Config
```

Comment:
Prefer not to share Configs between different clients, because they might change independently to require different things. Another reason for putting the two clients in separate packages.

Reply:
[b9825c4](https://github.com/Layr-Labs/eigenda-proxy/pull/301/commits/b9825c4e2bd7b72fc53b57584abd921d2940a36a)